python/tests/test_options.py

```diff
@@ -132,61 +245,267 @@ def test_argument_error_from_dict(self):
+        assert options.set_window_rect is True
+        assert options.unhandled_prompt_behavior == 'accept'
+        assert options.strict_file_interactability is True
+        assert options.timeouts == timeouts
 
     def test_sauce_options(self):
-        sauce = SauceOptions()
-        sauce.name = 'my-test'
-        sauce.record_screenshots = False
+        prerun = {'executable': 'http://url.to/your/executable.exe',
+                  'args': ['--silent', '-a', '-q'],
+                  'background': False,
+                  'timeout': 120}
+        custom_data = {'foo': 'foo',
+                       'bar': 'bar'}
+        tags = ['foo', 'bar']
+
+        options = SauceOptions()
+
+        options.avoid_proxy = True
+        options.build = 'Sample Build Name'
+        options.capture_performance = True
+        options.chromedriver_version = '71'
+        options.command_timeout = 2
+        options.custom_data = custom_data
+        options.extended_debugging = True
+        options.idle_timeout = 3
+        options.iedriver_version = '3.141.0'
+        options.max_duration = 300
+        options.name = 'Sample Test Name'
+        options.parent_tunnel = 'Mommy'
+        options.prerun = prerun
+        options.priority = 0
+        options.public = 'team'
+        options.record_logs = False
+        options.record_screenshots = False
+        options.record_video = False
+        options.screen_resolution = '10x10'
+        options.selenium_version = '3.141.59'
+        options.tags = tags
+        options.time_zone = 'San Francisco'
+        options.tunnel_identifier = 'tunnelname'
+        options.video_upload_on_pass = False
+
+        assert options.avoid_proxy is True
+        assert options.build == 'Sample Build Name'
+        assert options.capture_performance is True
+        assert options.chromedriver_version == '71'
+        assert options.command_timeout == 2
+        assert options.custom_data == custom_data
+        assert options.extended_debugging is True
+        assert options.idle_timeout == 3
+        assert options.iedriver_version == '3.141.0'
+        assert options.max_duration == 300
+        assert options.name == 'Sample Test Name'
+        assert options.parent_tunnel == 'Mommy'
+        assert options.prerun == prerun
+        assert options.priority == 0
+        assert options.public == 'team'
+        assert options.record_logs is False
+        assert options.record_screenshots is False
+        assert options.record_video is False
+        assert options.screen_resolution == '10x10'
+        assert options.selenium_version == '3.141.59'
+        assert options.tags == tags
+        assert options.time_zone == 'San Francisco'
+        assert options.tunnel_identifier == 'tunnelname'
+        assert options.video_upload_on_pass is False
+
+
+class TestAddingCapabilities(object):
+
+    def test_setting_capabilities(self):
+        with open(r'options.yml') as file:
+            yml = yaml.load(file)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of unsafe yaml load. Allows instantiation of arbitrary objects. Consider yaml.safe_load().](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4773242)